File: Trail of ones.py
class Solution:
    def numberOfConsecutiveOnes (ob,n):
        # code here 
        that, this = 0, 1
        
        for p in range(1, n-1):
            that, this = this, (that + this + pow(2, p, (10**9 + 7)))%(10**9 + 7)
            
        return this

File: Trail of ones.cpp
class Solution {
  public:
    
    int consec(int i, int prev,int n){
        if(i == n) return 1;
        
        int cnt = 0;
        
        cnt += consec(i+1, 0, n); //place 0
        
        if(prev == 0){
            cnt += consec(i+1, 1, n); //place 1
        }
        
        return cnt;
    }
    
    int countConsec(int n) {
        // code here
        int total = pow(2, n);
        int cons = consec(0,0,n);
        
        return total - cons;
    }
};

